framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java

```diff
@@ -81,7 +81,16 @@ public int getElementIndex() {
      */
     @Override
     public int getTypeCompoundIndex(Attribute.TypeCompound anno) {
-        return anno.getPosition().type_index;
+        int type_index = anno.getPosition().type_index;
+        // TODO: this is a temporary workaround of a bug in jsr308-langtools
```

Comment:
I'm not sure whether this is a jsr308 specific bug or a general one. The buggy code is located at com.sun.tools.javac.jvm.ClassReader in jsr308-langtools:
https://bitbucket.org/typetools/jsr308-langtools/src/71bc2f5e1d950f803437ae4c7c93ca22846b6fd8/src/share/classes/com/sun/tools/javac/jvm/ClassReader.java?at=default&fileviewer=file-view-default#ClassReader.java-1592

@wmdietl Prof. Dietl has reported this mismatch to Oracle a few weeks ago.